src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -797,22 +828,56 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_int_divide_int_overflow() {
+        let test_cases = vec![
+            (std::i64::MIN, false, -1, false),
+            (-1, false, 1, true),
+            (-2, false, 1, true),
+            (1, true, -1, false),
+            (2, true, -1, false),
+        ];
+        for (lhs, lhs_is_unsigned, rhs, rhs_is_unsigned) in test_cases {
+            let lhs_field_type = FieldTypeBuilder::new()
+                .tp(FieldTypeTp::LongLong)
+                .flag(if lhs_is_unsigned {
+                    FieldTypeFlag::UNSIGNED
+                } else {
+                    FieldTypeFlag::empty()
+                })
+                .build();
+            let rhs_field_type = FieldTypeBuilder::new()
+                .tp(FieldTypeTp::LongLong)
+                .flag(if rhs_is_unsigned {
+                    FieldTypeFlag::UNSIGNED
+                } else {
+                    FieldTypeFlag::empty()
+                })
+                .build();
+
+            let output: Result<Option<Int>> = RpnFnScalarEvaluator::new()
+                .push_param_with_field_type(lhs, lhs_field_type)
+                .push_param_with_field_type(rhs, rhs_field_type)
+                .evaluate(ScalarFuncSig::IntDivideInt);
+            assert!(output.is_err(), "lhs={:?}, rhs={:?}", lhs, rhs);
+        }
+    }
+
     #[test]
     fn test_int_divide_decimal() {
         let test_cases = vec![
-            ("11.01", "1.1", "10", false),
-            ("-11.01", "1.1", "-10", false),
-            ("11.01", "-1.1", "-10", false),
-            ("-11.01", "-1.1", "10", false),
-            ("123", "", "", false),
-            ("", "123", "", false),
+            ("11.01", "1.1", Some(10), false),
+            ("-11.01", "1.1", Some(-10), false),
+            ("11.01", "-1.1", Some(-10), false),
+            ("-11.01", "-1.1", Some(10), false),
+            ("123", "", None, false),
```

Comment:
Prefer to add Some to "123" as well, since this can also check "123" is successfully parsed.